src/wallet/wallet.cpp

```diff
@@ -626,6 +626,15 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
 
         Lock();
         Unlock(strWalletPassphrase);
+
+        // if we are using HD, replace the HD master key with a new one
+        if (!hdChain.masterKeyID.IsNull()) {
+            CKey key;
+            key.MakeNewKey(true);
+            if (!SetHDMasterKey(key))
```

Comment:
Agree that it is confusing to have `bool SomeFunction()`, where the return value indicates if something bad happens but at the same time this is unused and exceptions are thrown. This should be made consistent.